src/AvaloniaEdit/Search/SearchPanel.cs

```diff
@@ -111,7 +112,7 @@ public string SearchPattern
         public bool IsReplaceMode
         {
             get => GetValue(IsReplaceModeProperty);
-            set => SetValue(IsReplaceModeProperty, value);
+            set => SetValue(IsReplaceModeProperty, _textEditor?.IsReadOnly ?? false ? false : value);
```

Comment:
AvaloniaProperties should not have additional code in their getters/setters since they won't necessarily be called (e.g. when using binding, animation, etc.) . You should place this in the value changed callback, set in `AvaloniaProperty.Register`.

Reply:
I tried to do that and while the value stays correct the replace window still opens. I don't know why it is not working